flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -512,6 +528,33 @@ class GroupedTableImpl(
   }
 }
 
+/**
+  * The implementation of an [[AggregatedTable]] that has been performed on an aggregate function.
+  */
+class AggregatedTableImpl(
+    private[flink] val table: Table,
+    private[flink] val groupKeys: Seq[Expression],
+    private[flink] val aggregateFunction: Expression)
+  extends AggregatedTable {
+
+  private val tableImpl = table.asInstanceOf[TableImpl]
+
+  override def select(fields: String): Table = {
+    select(ExpressionParser.parseExpressionList(fields).asScala: _*)
+  }
+
+  override def select(fields: Expression*): Table = {
+    new TableImpl(tableImpl.tableEnv,
+      tableImpl.operationTreeBuilder.project(fields,
```

Comment:
I think we can unify aggregate and flatAggregate implementations. i.e. both `AggregatedTableImpl#select` and `FlatAggregateTableImpl#select` can using`operationTreeBuilder.project` or `xxTable.select(fields: _*)`. What do you think?